contracts/registry/src/registry.rs

```diff
@@ -356,28 +385,48 @@ impl SBTRegistry for Contract {
         self.supply_by_owner
             .insert(supply_key, &(old_supply_to + tokens_recovered));
 
-        // ---
-        // add old_owner to a bannded list
-        self.banlist.insert(&from);
-
-        // emit Recover event
-        SbtRecover {
-            issuer: &issuer,
-            old_owner: &from,
-            new_owner: &to,
+        let completed = tokens_recovered != limit as u64;
+        if completed {
+            if resumed {
+                // insert is happening when we need to continue, so don't need to remove if
+                // the process finishes in the same transaction.
+                self.ongoing_soul_tx.remove(&from);
+            }
+            // we emit the event only once the operation is completed and only if some tokens were
+            // transferred
+            if resumed || tokens_recovered > 0 {
+                // add old_owner to a bannded list
+                self.banlist.insert(&from);
```

Comment:
we need to ban him when starting the process, otherwise a user can do `sbt_soul_transfer` when an issuer is doing recovery.

Reply:
DONE